sherpa/conftest.py

```diff
@@ -544,54 +549,36 @@ def old_numpy_printing():
         np.set_printoptions(legacy=oldopts['legacy'])
 
 
-PLOT_BACKENDS = [dummy_backend]
-if pylab_backend is not None:
-    PLOT_BACKENDS.append(pylab_backend)
-
-
-@pytest.fixture(params=PLOT_BACKENDS)
-def override_plot_backend(request):
+@pytest.fixture(params=ALL_PLOT_BACKENDS)
+def all_plot_backends(request):
     """Override the plot backend for this test
 
     Runs the test with the given plot backend and then restores the
     original value.
 
-    Note that this does not reload the ui layers, which will have
+    Note that it is not clear at this point if
+    this does or does not reload the ui layers, which will have
     retain reference to the original backend throughout.
 
     """
+    with TemporaryPlottingBackend(request.param):
+        yield
 
-    old = sherpa.plot.backend
-
-    changed = old.name != request.param.name
-    if changed:
-        sherpa.plot.backend = request.param
-        sherpa.astro.plot.backend = request.param
 
-    yield
-    if changed:
-        sherpa.plot.backend = old
-        sherpa.astro.plot.backend = old
+@pytest.fixture(params=PLOT_BACKENDS)
```

Comment:
This has lost

    @pytest.fixture(autouse=True, scope="session") 

which means that we now require all tests that could create a matplotlib plot to be marked with this decorator. I am concerned that it is easy to miss a test. However, I guess that this would be picked up by the CI run (and it only matters if you add a test that is run after the last test marked with `plot_backends` is run.

On the plus side, it should lead to less memory being taken up by old matplotlib plots (as it's done per test).

Reply:
I can confirm that CI tells me if matplotlib is needed, but it's not marked with a decorator; this PR has a long line of CI failures before I got them all ;-)